### PR TITLE
replaced the deprecated findAndModify method from native js driver to…

### PR DESCRIPTION
**Pull request by lushc**
… findOneAndUpdate

Closes #268

### CODE REVIEW

**Comment by simison** [comment]
Node.js v0.11 failed tests — how about we just ditch v0.10 and v0.11 going forward?

**Comment by lushc** [comment]
I'm not entirely sure why it failed and if it's down to the Node version, [the native driver still supports v0.10 and v0.12](https://github.com/mongodb/node-mongodb-native/blob/2.2/.travis.yml), so I'm assuming it was just an intermittent connection issue ([they mention this in the docs](https://docs.travis-ci.com/user/database-setup/#MongoDB-does-not-immediately-accept-connections)) which backs up the actual build error.

**Comment by simison** [comment]
@lushc if you wanna debug/test that, you can close+re-open to re-run Travis.

**Comment by lushc** [comment]
Ah, that'd be much quicker than testing it in a TravisCI docker container like I just did...

**Comment by lushc** [comment]
@simison all passing this time, rather than removing these versions do you think we should mimic the native driver and test against 0.12 instead of 0.11? Although they have been end-of-life since late 2016, so maybe removing them is better, not sure.

**Comment by simison** [comment]
0.12 might just pass or bounce again randomly, I'm not sure if it's really worth effort. They're end-of-life for exactly this reason that people wouldn't need to keep supporting them as it's time consuming. ;-)

**Comment by simison** [comment]
...I'd merge this without rebasing and then merge #449 later if no-one has anything against it.

**Comment by lushc** [comment]
Oh I didn't remove 0.11 by the way, so #449 can be merged whenever. The build failure was just an intermittent issue where Mongo wasn't ready by the time the tests started running, which is a bit odd as the build already waits 15 seconds for it. This has happened to me before on other projects so I'm probably cursed :P

**Comment by simison** [comment]
@lushc apparently the same issue with Node.js v0.11 occurred here: https://github.com/agenda/agenda/pull/331#issuecomment-221035610

**Comment by simison** [comment]
Thanks a bunch btw for this! Happy to get some actual code merged in finally. :-)

**Comment by lushc** [comment]
No problem, you've really managed to get the ball rolling on triaging issues/PRs and getting an active core of maintainers together, so thank you 👍 🍺  a remarkable difference compared to early last week!

File: lib/agenda.js [hunk 1]
```diff
@@ -427,17 +427,16 @@ Agenda.prototype._findAndLockNextJob = function(jobName, definition, cb) {
...
         $or: [
           {name: jobName, lockedAt: null, nextRunAt: {$lte: this._nextScanAt}, disabled: { $ne: true }},
           {name: jobName, lockedAt: {$exists: false}, nextRunAt: {$lte: this._nextScanAt}, disabled: { $ne: true }},
           {name: jobName, lockedAt: {$lte: lockDeadline}, disabled: { $ne: true }}
         ]
       },
-      {'priority': -1},  // sort
       {$set: {lockedAt: now}},  // Doc
-      {'new': true},  // options
+      {returnOriginal: false, 'priority': -1},  // options & sort
```
Comment (lushc) [inline lib/agenda.js]: I've just noticed that according to the [findOneAndUpdate docs](http://mongodb.github.io/node-mongodb-native/2.2/api/Collection.html#findOneAndUpdate) this is incorrect. It should be:

```js
{returnOriginal: false, sort: {priority: -1}}
```

There doesn't seem to be any test coverage for this either. I'll put in another PR to fix.

Comment (simison) [inline lib/agenda.js]: Follow ups https://github.com/agenda/agenda/pull/451

**Pull request merged**